### PR TITLE
Replace PR bot's diff tool with one that supports pagination

### PR DESCRIPTION
**Pull request by jdisanti**
## Motivation and Context
A codegen diff that doesn't get truncated would be really helpful with diagnosing how my changes in #2129 broke the endpoint tests, so I spent a few hours building a better diff tool for PR bot to address #1219. This also allows us to remove Node from the Docker build image.

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: tools/difftags/src/page.rs [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+ 29
+ 30
+        "#;
+        let mut patch = PatchSet::new();
+        patch.parse(&diff_str).unwrap();
+
+        let hunk = &patch.files()[0].hunks()[0];
+        let section: Section = hunk.into();
+
+        assert_eq!((1, 1), section.start_line);
```
Comment (ysaito1001) [inline tools/difftags/src/page.rs:224]: Can we explain in the comment how the the fields in `Section` are derived from the test data above? I can sort of guess from the code and `assert_eq!` but am not 100% sure, e.g., how the `section.diff` starts at `05` and ends at `25`.

**Review by ysaito1001 — APPROVED** [review]

File: tools/codegen-diff-revisions.py [hunk 2]
```diff
@@ -120,61 +120,27 @@ def generate_and_commit_generated_code(revision_sha):
...
         return None
     else:
-        run(f"mkdir -p {OUTPUT_PATH}/{base_commit_sha}/{head_commit_sha}")
-        dest_path = f"{base_commit_sha}/{head_commit_sha}/diff-{suffix}.html"
+        partial_output_path = f"{base_commit_sha}/{head_commit_sha}/{suffix}"
+        full_output_path = f"{OUTPUT_PATH}/{partial_output_path}"
+        run(f"mkdir -p {full_output_path}")
+        run(f"git diff --output=codegen-diff.txt -U30 {whitespace_flag} {BASE_BRANCH_NAME} {HEAD_BRANCH_NAME} -- {path_to_diff}")
+
+        # Generate HTML diff. This uses the diff2html-cli, which defers to `git diff` under the hood.
```
[the remark this answers is not in the document]
Comment (jdisanti) [reply tools/codegen-diff-revisions.py]: Nope, I just failed to update the comment 😄

File: tools/difftags/src/page.rs (hunk 1 above)
Comment (jdisanti) [reply tools/difftags/src/page.rs:231]: Yeah, good call. I created a constant for the number of lines surrounding the modified lines that will be displayed, commented it, and also added additional context to the assert messages.

**Review by ysaito1001 — APPROVED** [review]
12bcfc8, efa3b61 look good!

**Pull request merged**